app/lib/dns.server.ts

```diff
@@ -157,8 +158,30 @@ export const getChangeStatus = async (changeId: string) => {
   }
 };
 
-const isNameValid = (name: string) => {
-  return /^[a-z0-9-]+.[a-z0-9-]+.[a-z]+.[a-z]+\.?$/.test(name);
+//Domain name rules
+//1. Domain name pattern should be [name].[studentId].rootDomain.com(.)
+//2. Domain name can contain only alphanumerical characters, '-', and '_'
+//3. Domain name should not start or end with -
+//4. Domain name cannot contain multiple consecutive '-' or '_'
+//5. Domain name can contain uppercase in UI but it is converted to lowercase before validation
+//6. studentId only contains alphanumerical characters
+export const isNameValid = (name: string) => {
+  const rootDomain: string = process.env.ROOT_DOMAIN!;
+  let subdomain: string;
+
+  //It only validates domain name and studentId, not root domain
+  if (name.endsWith('.')) {
+    subdomain = name.replace(`.${rootDomain}.`, '');
+  } else {
+    subdomain = name.replace(`.${rootDomain}`, '');
+  }
+  return (
+    /^(?!.*[-_]{2,})(?!^[-])[a-z0-9_-]+[^-_]\.[a-z0-9]+$/.test(subdomain) &&
+    validator.isFQDN(name, {
+      allow_underscores: true,
+      allow_trailing_dot: true,
```

Comment:
I think this should be false, we should not allow both variants IMO

Reply:
There was a suggestion to add trailing dot. I think it's because Route 53 automatically adds a dot somewhere. But it would very unlikely get `root domain` from Route 53. 

I thought you would need underscore for `_acme...`

Reply:
yes, I need the underscore, but I'd not use the trailing dot. If Route53 needs it, I'd add it at the invocation of route53 and abstract away this complexity from the rest of the system